### PR TITLE
Enable Micronaut HyperlinkProviders - deadlocks fixed.

### PR DESCRIPTION
**Pull request by dbalek**
Enable Micronaut HyperlinkProviders again. Deadlocks fixed - `ParserManager.parse(...)` should not be called from AWT EDT anymore.

### CODE REVIEW

**Review by MartinBalin — APPROVED** [review]

File: enterprise/micronaut/src/org/netbeans/modules/micronaut/MicronautConfigUtilities.java [hunk 1]
```diff
@@ -211,6 +194,70 @@ public static void collectUsages(FileObject fo, String propertyName, Consumer<Us
...
+            Language language = LanguageRegistry.getInstance().getLanguageByMimeType(r.getSnapshot().getMimeType());
+            if (language != null) {
+                StructureScanner scanner = language.getStructure();
+                if (scanner != null) {
+                    Map<String, ConfigurationMetadataGroup> groups = MicronautConfigProperties.getGroups(project);
+                    scan(scanner.scan((ParserResult) r), new Stack<>(), context -> {
+                        if (!context.empty()) {
+                            String propertyName = getPropertyName(context);
+                            List<ConfigurationMetadataSource> sources = new ArrayList<>();
+                            ConfigurationMetadataProperty property = getProperty(groups, propertyName, sources);
```
Comment (sdedic) [inline enterprise/micronaut/src/org/netbeans/modules/micronaut/MicronautConfigUtilities.java:210]: Does this depend on the scanning context ? If not, pls. move outside of the lambda - less invocations.

File: enterprise/micronaut/src/org/netbeans/modules/micronaut/hyperlink/MicronautConfigHyperlinkProvider.java [hunk 2]
```diff
@@ -42,28 +44,41 @@
...
- *
+ * 
  * @author Dusan Balek
  */
 public class MicronautConfigHyperlinkProvider implements HyperlinkProviderExt {
 
-    //@MimeRegistration(mimeType = "text/x-yaml", service = HyperlinkProviderExt.class, position = 1250)
+    private static final String SPANS_PROPERTY_NAME = "MicronautConfigHyperlinkSpans";
+
+    @MimeRegistration(mimeType = "text/x-yaml", service = HyperlinkProviderExt.class, position = 1250)
```
Comment (sdedic) [inline enterprise/micronaut/src/org/netbeans/modules/micronaut/hyperlink/MicronautConfigHyperlinkProvider.java:76]: nitpick: constant for YAML MIME type?

File: enterprise/micronaut/src/org/netbeans/modules/micronaut/hyperlink/MicronautConfigHyperlinkProvider.java [hunk 3]
```diff
@@ -80,10 +95,26 @@ public boolean isHyperlinkPoint(Document doc, int offset, HyperlinkType type) {
...
-        return property != null || !sources.isEmpty() ? span : null;
+        String mimeType = DocumentUtilities.getMimeType(doc);
+        if ("text/x-yaml".equals(mimeType)) {
+            List<int[]> spans = null;
+            synchronized (doc) {
+                spans = (List<int[]>) doc.getProperty(SPANS_PROPERTY_NAME);
+            }
+            if (spans != null) {
+                for (int[] span : spans) {
+                    if (span.length == 2 && span[0] <= offset && offset <= span[1]) {
```
Comment (sdedic) [inline enterprise/micronaut/src/org/netbeans/modules/micronaut/hyperlink/MicronautConfigHyperlinkProvider.java:106]: Maybe `< span[1]` (exclusive) ?

**Review by sdedic — APPROVED** [review]

File: enterprise/micronaut/src/org/netbeans/modules/micronaut/hyperlink/MicronautConfigHyperlinkProvider.java (hunk 3 above)
Comment (dbalek) [reply enterprise/micronaut/src/org/netbeans/modules/micronaut/hyperlink/MicronautConfigHyperlinkProvider.java:106]: Go to declaration usually works even if caret is placed at the identifier end position `someId|`

**Pull request merged**